### PR TITLE
Feature/vta 404

### PR DESCRIPTION
**Pull request by IuliaDVSA**
Fixes on remaining expired dates on BE regression pack

### CODE REVIEW

File: src/test/java/testresults/PutTestResultsCertificateNotGenerated.java [hunk 1]
```diff
@@ -136,7 +136,8 @@ public PutTestResultsCertificateNotGenerated(String testResult, String testStatu
 
     }
 
-    @WithTag("annual_certificates")
+    //@WithTag("annual_certificates")
```
Comment (cb-cs) [inline src/test/java/testresults/PutTestResultsCertificateNotGenerated.java:139]: Is the commented out code required?

File: src/test/java/testresults/annual_certificates/TestPostTestResultsAnnualCertificateGenerationHgv.java [hunk 2]
```diff
@@ -241,7 +242,8 @@ public TestPostTestResultsAnnualCertificateGenerationHgv(String name, String tes
         this.testCode = testCode;
     }
 
-    @WithTag("annual_certificates")
+//    @WithTag("annual_certificates")
```
Comment (cb-cs) [inline src/test/java/testresults/annual_certificates/TestPostTestResultsAnnualCertificateGenerationHgv.java:245]: Same here, is commented out code required?

File: src/test/java/testresults/annual_certificates/TestPostTestResultsAnnualCertificateGenerationTrl.java [hunk 3]
```diff
@@ -299,7 +299,11 @@ public void testResults_Annual_Certificate_Generation_Trl() {
         // Post the results, together with any alterations, and verify that they are accepted.
         testResultsSteps.postVehicleTestResultsWithAlterations(testResultRecord, alterations);
         if ("warmup test".equals(name)) {
-            //ignoring results
+            try {
```
Comment (cb-cs) [inline src/test/java/testresults/annual_certificates/TestPostTestResultsAnnualCertificateGenerationTrl.java:302]: Can't run the code locally, so can you confirm please. Is an exception actually thrown here if statusCode is not 201?

File: src/test/java/vehicle/PutVehicleTechnicalRecords.java [hunk 4]
```diff
@@ -2113,7 +2113,8 @@ public void testPostVehicleTechRecordTrlPrsNotifiableAlteration(){
         vehicleTechnicalRecordsSteps.valueForFieldInPathShouldContains("[0].techRecord.find { it.statusCode == 'current' }.createdAt", new SimpleDateFormat("yyyy-MM-dd").format(date));
     }
 
-    @WithTag("Vtm")
+//    @WithTag("Vtm")
```
Comment (cb-cs) [inline src/test/java/vehicle/PutVehicleTechnicalRecords.java:2116]: Same here, is commented out code required?

File: src/test/java/testnumber/PostTestNumber.java [hunk 5]
```diff
@@ -14,6 +14,7 @@
 import net.thucydides.core.annotations.Title;
 import net.thucydides.core.annotations.WithTag;
 import org.apache.commons.lang3.RandomStringUtils;
+import org.junit.Before;
```
Comment (m-coslett) [inline src/test/java/testnumber/PostTestNumber.java:17]: please remove unused imports

File: src/test/java/testresults/PostTestResultsApiSpecsVehicleConfigurationHgv.java [hunk 6]
```diff
@@ -8,6 +8,7 @@
 import net.serenitybdd.junit.runners.SerenityParameterizedRunner;
 import net.thucydides.core.annotations.Steps;
 import net.thucydides.core.annotations.Title;
+import net.thucydides.core.annotations.WithTag;
```
Comment (m-coslett) [inline src/test/java/testresults/PostTestResultsApiSpecsVehicleConfigurationHgv.java]: Please remove unused import

File: src/test/java/testresults/PostTestResultsMoreVinFirstTest.java [hunk 7]
```diff
@@ -5,6 +5,7 @@
 import net.serenitybdd.junit.runners.SerenityRunner;
 import net.thucydides.core.annotations.Steps;
 import net.thucydides.core.annotations.Title;
+import net.thucydides.core.annotations.WithTag;
```
Comment (m-coslett) [inline src/test/java/testresults/PostTestResultsMoreVinFirstTest.java]: Please remove unused import

File: src/test/java/testresults/PostTestResultsMoreVinNotifiableAlteration.java [hunk 8]
```diff
@@ -17,6 +17,7 @@
 import java.util.List;
 import java.util.UUID;
 
+
```
Comment (m-coslett) [inline src/test/java/testresults/PostTestResultsMoreVinNotifiableAlteration.java:18]: Please remove unused import

File: src/test/java/testresults/PostTestResultsPozMainLvlCancelled.java [hunk 9]
```diff
@@ -7,6 +7,7 @@
 import net.serenitybdd.junit.runners.SerenityRunner;
 import net.thucydides.core.annotations.Steps;
 import net.thucydides.core.annotations.Title;
+import net.thucydides.core.annotations.WithTag;
```
Comment (m-coslett) [inline src/test/java/testresults/PostTestResultsPozMainLvlCancelled.java]: Please remove unused import

File: src/test/java/testresults/PostTestResultsProvisionalUpdate.java [hunk 10]
```diff
@@ -35,6 +35,7 @@ public void beforeTest() {
         this.testStartDate = LocalDateTime.now();
     }
 
+
```
Comment (m-coslett) [inline src/test/java/testresults/PostTestResultsProvisionalUpdate.java:37]: Please remove unused import

File: src/test/java/testresults/PostTestResultsProvisionalUpdateFirstTestHgv.java [hunk 11]
```diff
@@ -5,6 +5,7 @@
 import net.serenitybdd.junit.runners.SerenityParameterizedRunner;
 import net.thucydides.core.annotations.Steps;
 import net.thucydides.core.annotations.Title;
+import net.thucydides.core.annotations.WithTag;
```
Comment (m-coslett) [inline src/test/java/testresults/PostTestResultsProvisionalUpdateFirstTestHgv.java]: Please remove unused import

File: src/test/java/testresults/PostTestResultsRoadworthinessCertificateGeneration.java [hunk 12]
```diff
@@ -5,6 +5,7 @@
 import net.serenitybdd.junit.runners.SerenityRunner;
 import net.thucydides.core.annotations.Steps;
 import net.thucydides.core.annotations.Title;
+import net.thucydides.core.annotations.WithTag;
```
Comment (m-coslett) [inline src/test/java/testresults/PostTestResultsRoadworthinessCertificateGeneration.java]: Please remove unused import

File: src/test/java/testresults/PostTestResultsSystemNumber.java [hunk 13]
```diff
@@ -6,6 +6,7 @@
 import net.serenitybdd.junit.runners.SerenityRunner;
 import net.thucydides.core.annotations.Steps;
 import net.thucydides.core.annotations.Title;
+import net.thucydides.core.annotations.WithTag;
```
Comment (m-coslett) [inline src/test/java/testresults/PostTestResultsSystemNumber.java]: Please remove unused import

File: src/test/java/testresults/PostTestResultsTestCodeMappingOnTestTypes.java [hunk 14]
```diff
@@ -8,6 +8,7 @@
 import net.serenitybdd.junit.runners.SerenityRunner;
 import net.thucydides.core.annotations.Steps;
 import net.thucydides.core.annotations.Title;
+import net.thucydides.core.annotations.WithTag;
```
Comment (m-coslett) [inline src/test/java/testresults/PostTestResultsTestCodeMappingOnTestTypes.java]: Please remove unused import

File: src/test/java/testresults/PutTestResultsCertificateNotGenerated.java (hunk 1 above)
Comment (IuliaDVSA) [reply src/test/java/testresults/PutTestResultsCertificateNotGenerated.java]: Removed

File: src/test/java/testresults/annual_certificates/TestPostTestResultsAnnualCertificateGenerationHgv.java (hunk 2 above)
Comment (IuliaDVSA) [reply src/test/java/testresults/annual_certificates/TestPostTestResultsAnnualCertificateGenerationHgv.java]: Removed

File: src/test/java/testresults/PostTestResultsTestCodeMappingOnTestTypes.java (hunk 14 above)
Comment (IuliaDVSA) [reply src/test/java/testresults/PostTestResultsTestCodeMappingOnTestTypes.java]: Removed

Comment (IuliaDVSA) [reply src/test/java/testresults/PostTestResultsTestCodeMappingOnTestTypes.java]: Removed

File: src/test/java/testresults/PostTestResultsSystemNumber.java (hunk 13 above)
Comment (IuliaDVSA) [reply src/test/java/testresults/PostTestResultsSystemNumber.java]: Removed

File: src/test/java/testresults/PostTestResultsRoadworthinessCertificateGeneration.java (hunk 12 above)
Comment (IuliaDVSA) [reply src/test/java/testresults/PostTestResultsRoadworthinessCertificateGeneration.java]: Removed

File: src/test/java/testresults/PostTestResultsProvisionalUpdateFirstTestHgv.java (hunk 11 above)
Comment (IuliaDVSA) [reply src/test/java/testresults/PostTestResultsProvisionalUpdateFirstTestHgv.java]: Removed

File: src/test/java/testresults/PostTestResultsProvisionalUpdate.java (hunk 10 above)
Comment (IuliaDVSA) [reply src/test/java/testresults/PostTestResultsProvisionalUpdate.java:37]: Removed

File: src/test/java/testresults/PostTestResultsPozMainLvlCancelled.java (hunk 9 above)
Comment (IuliaDVSA) [reply src/test/java/testresults/PostTestResultsPozMainLvlCancelled.java]: Removed

File: src/test/java/testresults/PostTestResultsMoreVinNotifiableAlteration.java (hunk 8 above)
Comment (IuliaDVSA) [reply src/test/java/testresults/PostTestResultsMoreVinNotifiableAlteration.java:18]: Removed

File: src/test/java/testresults/PostTestResultsMoreVinFirstTest.java (hunk 7 above)
Comment (IuliaDVSA) [reply src/test/java/testresults/PostTestResultsMoreVinFirstTest.java]: Removed

File: src/test/java/testnumber/PostTestNumber.java (hunk 5 above)
Comment (IuliaDVSA) [reply src/test/java/testnumber/PostTestNumber.java:17]: Removed

File: src/test/java/vehicle/PutVehicleTechnicalRecords.java (hunk 4 above)
Comment (IuliaDVSA) [reply src/test/java/vehicle/PutVehicleTechnicalRecords.java]: Removed

Comment (IuliaDVSA) [reply src/test/java/vehicle/PutVehicleTechnicalRecords.java]: Restored

File: src/test/java/testresults/annual_certificates/TestPostTestResultsAnnualCertificateGenerationTrl.java (hunk 3 above)
Comment (IuliaDVSA) [reply src/test/java/testresults/annual_certificates/TestPostTestResultsAnnualCertificateGenerationTrl.java:303]: Added a sout to print out exception.

File: src/test/java/testresults/annual_certificates/TestPostTestResultsAnnualCertificateGenerationHgv.java [hunk 15]
```diff
@@ -277,6 +278,13 @@ public void testResults_Annual_Certificate_Generation_Hgv() {
 
         // Post the results, together with any alterations, and verify that they are accepted.
         testResultsSteps.postVehicleTestResultsWithAlterations(testResultRecord, alterations);
+        if ("warmup test".equals(name)) {
```
Comment (m-coslett) [inline src/test/java/testresults/annual_certificates/TestPostTestResultsAnnualCertificateGenerationHgv.java:281]: as per Craig's query, does this throw an exception if the status response is not 201 or is it just an assertion fail? We cant run this locally so if this is working as you expect, please disregard these comments

Comment (IuliaDVSA) [reply src/test/java/testresults/annual_certificates/TestPostTestResultsAnnualCertificateGenerationHgv.java:281]: Ran it locally and the tests, pass, added a sout to print out a Retry and exception if it returns anything other than 201

Comment (IuliaDVSA) [reply src/test/java/testresults/annual_certificates/TestPostTestResultsAnnualCertificateGenerationHgv.java:281]: Added sout on all 3 classes.

**Review by m-coslett — APPROVED** [review]

**Review by cb-cs — APPROVED** [review]

**Pull request merged**